File: frontend/src/game/upgrades/stat_modifiers/player_speed_1.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerSpeed1Def: UpgradeDef = {
  id: "player_speed_1",
  name: "Thruster Boost",
  description: "+10 movement speed",
  rarity: RarityID.Uncommon,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 6,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.Speed,
  value: 10,
  stackable: true,
  maxStacks: 10,
}

export class PlayerSpeed1 extends Upgrade {}

File: backend/app/core/upgrades/stat_modifiers/player_speed_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerSpeed1(UpgradeImplementation):
    id = 'player_speed_1'
    name = 'Thruster Boost'
    description = '+10 movement speed'
    rarity = 'uncommon'
    type = 'stat_modifier'
    cost = 6

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 10
        appliers.modifiers.add_multiplier('player', 'speed', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True
